packages/template-retail-react-app/app/pages/product-detail/index.test.js

```diff
@@ -160,3 +161,90 @@ describe('product set', () => {
         })
     })
 })
+
+describe('product bundles', () => {
+    beforeEach(() => {
+        global.server.use(
+            // Use product bundle instead of product set
+            rest.get('*/products/:productId', (req, res, ctx) => {
+                return res(ctx.delay(0), ctx.status(200), ctx.json(mockProductBundle))
+            }),
+            // For adding items to basket
+            rest.post('*/baskets/:basketId/items', (req, res, ctx) => {
+                return res(ctx.json(basketWithProductSet))
+            })
+        )
+    })
+
+    test('renders multi-product layout', async () => {
+        renderWithProviders(<MockedComponent />)
+
+        await waitFor(
+            () => {
+                expect(screen.getAllByTestId('product-view')).toHaveLength(4) // 1 parent + 3 children
+            },
+            {timeout: 5000}
+        )
+    })
+
+    test('add the bundle to cart successfully', async () => {
+        const urlPathAfterSelectingAllVariants =
+            '/en-GB/product/test-bundle?25592770M=color%3DJJGN9A0%26size%3D006&25565139M=color%3DJJ169XX%26size%3D9SM&25565094M=color%3DJJ0CZXX%26size%3D9XS'
+        window.history.pushState({}, 'ProductDetail', urlPathAfterSelectingAllVariants)
+
+        // Initial basket is necessary to add items to it
+        const initialBasket = {basketId: 'valid_id'}
+        renderWithProviders(<MockedComponent />, {wrapperProps: {initialBasket}})
+
+        await waitFor(
+            () => {
+                expect(screen.getAllByText("Women's clothing test bundle")[0]).toBeInTheDocument()
+            },
+            {timeout: 5000}
+        )
+
+        const buttons = await screen.findAllByText(/add bundle to cart/i)
+        fireEvent.click(buttons[0])
+
+        await waitFor(
+            () => {
+                const modal = screen.getByTestId('add-to-cart-modal')
+                expect(within(modal).getByText(/1 item added to cart/i)).toBeInTheDocument()
+            },
+            // Seems like rendering the modal takes a bit more time
+            {timeout: 10000}
+        )
+    }, 30000)
```

Comment:
This test timeout may no longer be necessary, since we've already set the global timeout to 30,000 ms for CI: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/b7c0aafcdd5eb88e986db039b82bcbca281454ff/packages/template-retail-react-app/jest.config.js#L48-L49